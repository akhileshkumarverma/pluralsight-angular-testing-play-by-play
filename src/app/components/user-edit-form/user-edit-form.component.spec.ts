import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA, SimpleChanges } from '@angular/core';

import { UserEditFormComponent } from './user-edit-form.component';
declare var toMatchSnapshot: any;

describe('UserEditFormComponent', () => {
  let component: UserEditFormComponent;
  let fixture: ComponentFixture<UserEditFormComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [UserEditFormComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserEditFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should render markup from snapshot`, () => {
    expect(fixture).toMatchSnapshot();
  });

  it(`should have valid form`, () => {
    const fakeUser = { id: 1, name: 'fake' };
    const changes = {user:{currentValue: fakeUser}} as any;
    component.ngOnChanges(changes);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
