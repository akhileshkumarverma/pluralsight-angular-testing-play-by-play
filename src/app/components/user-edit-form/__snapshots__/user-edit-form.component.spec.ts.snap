// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEditFormComponent should have valid form 1`] = `
<app-user-edit-form
  save={[Function EventEmitter]}
  userForm={[Function FormGroup]}
>
  <form
    novalidate=""
  >
    
      
    <input
      formcontrolname="name"
      type="text"
    />
    
      
    <button
      type="submit"
    >
      Submit
    </button>
    
      
    
    
  
  </form>
</app-user-edit-form>
`;

exports[`UserEditFormComponent should render markup from snapshot 1`] = `
<app-user-edit-form
  save={[Function EventEmitter]}
  userForm={[Function FormGroup]}
>
  <form
    novalidate=""
  >
    
      
    <input
      formcontrolname="name"
      type="text"
    />
    
      
    <button
      type="submit"
    >
      Submit
    </button>
    
      
    
    <p>
      
          Name is required
      
    </p>
    
  
  </form>
</app-user-edit-form>
`;
